### DISCUSSION TITLE
Microcontroller cannot connect when USB to UART module is already connected

### DISCUSSION THREAD

**Discussion by vocai-bar** [micropython/micropython#15736] [RP2040 / Pico]
Hi,
I've connected my Adafruit RP2040 QT PY to the CH340E USB to UART module.
When I'm running a code for printing messages to the UART, its working and I see what I've printed.
The problem occurs when the CH340E is already connected to the USB and the RX and TX legs, it says USB device is not recognize and the microcontroller is not running and not sending messages.
I tried to click the reset button and it didn't work for communicating with the UART but it did work for turning on LED when the CH340E is already connected.
I've also  tried to connect the CH340E to the USB and connect the microcontroller to a separate power source and it still didn't work.

Any suggestions beside connecting the microcontroller before the UART module?

Thanks.

**Comment by vocai-bar** [comment]
I figure out that the problem is that I've used A0 and A1 for RX and TX communication. in the other RX/TX with UART1 its working fine.
The problem is that i need to connect those specific legs.
Someone have any explanation for that? There are any other ways to connect in those GPIO's?

Pinout of the micro controller
![image](https://github.com/user-attachments/assets/9ee1dd85-0206-46cb-a0df-045085b59929)

**Comment by robert-hh** [comment]
I do not understand what you want to achieve. I understand that you can connect a CH340 to A0/A1 and use it with MicroPython as UART(0). Depending on the configuration, you can as well have the REPL prompt at that UART. If you connect both the CH340 USB and the QT_PY's USB to you PC, you may see two different USB devices. Does that happen?
With Linux, run `lsusb`. With Windows, look into the device manager.
If you use Thonny or another information hiding IDE, switch it then to the proper device.

**Reply by vocai-bar** [reply]
Let me try to re-explain the problem I'm facing.
When the CH340 connected first to the USB and then I'm trying to connect the other USB/Other power connection the it wont connect - it says "USB device is not recognize" in the computer. But when I connect in the opposite order - Microcontroller first and then CH340 its working and I can see 2 devices in the device manager.
This problem occur only in the GPIO 28, 29 in UART(0) where there are analog connection aswell. in the other GPIO's with UART(1) its working just fine.
The issue is that I need those connected because the board that I've design support it, and also I'm using UART(1) to connect another component.

Thanks

**Reply by robert-hh** [reply]
Which OS runs on your PC? Windows? Linux? MAC? Are you connecting to the same Virtual OS port all the time? Which tool do you use to connect?

**Reply by robert-hh** [reply]
Do you connect RX, TX and GND to the CH340 device?

**Reply by vocai-bar** [reply]
Hi,
Im running on windows. To view the connection im using Putty and to run the program I'm using Thonny.
And yes, RX,TX and GND all connected to the CH340, all the grounds are connected (from the power supply aswell if USB is'nt connected to the microcontroller)

So just to be clear to run this program im connecting the running the code of the microcontroller through USB in thonny (windows OS) and then i disconnect it and connect it to a different power supply  and connecting all the grounds together, while the CH340 is connect via USB. RX, TX and GND are connected from the microcontroller to the CH340 device.


Thanks for helping.

**Reply by vocai-bar** [reply]
> Hi, Im running on windows. To view the connection im using Putty and to run the program I'm using Thonny. And yes, RX,TX and GND all connected to the CH340, all the grounds are connected (from the power supply aswell if USB is'nt connected to the microcontroller)
> 
> So just to be clear to run this program im connecting the running the code of the microcontroller through USB in thonny (windows OS) and then i disconnect it and connect it to a different power supply and connecting all the grounds together, while the CH340 is connect via USB. RX, TX and GND are connected from the microcontroller to the CH340 device.
> 
> Thanks for helping.

any other suggestions?
